### ISSUE TITLE
Rethink fencing for Kokkos::resize

### ISSUE THREAD

**Issue by masterleinad** [kokkos/kokkos#6186]
If we're thinking about this like allocation + `deep_copy`, don't we need a global fence before this remap call, and then only a local fence after? We can do the allocation and copy stream-wise and fence them locally, but we don't know what stream may have been modifying the data in `v` before it got passed in.

_Originally posted by @PhilMiller in https://github.com/kokkos/kokkos/pull/6184#discussion_r1218774809_

**Comment by ajpowelsnl** [comment]
@masterleinad and @PhilMiller - should we discuss this issue on Wed.?

**Comment by masterleinad** [comment]
No, the issues resulted from a discussion for the hotfix proposed in #6184.
